### PR TITLE
keyboard bindings from rhai scripts

### PR DESCRIPTION
**Pull request by dragonnn**
draft work for keyboard bindings from rhai scripts. Example of binding a combination:
```
keyboard::register(|| {
    log::info("key called H");
}, "Super_L", ["h"]);
```
also adds log and system module for rhai.
Future plans:
- add access to current state of Anodium in callbacks
- implement sending events to callloop to allow switch workspaces, tty etc. from rhai scripts
- add extra parameter to keyboard::register with allows to capture all numeric/ascii keys after a combination and pass the pressed keys too callback (for example for doing a combo Super_L + Shift + [1-9] to switch workspaces)

### CODE REVIEW

File: src/config/keyboard.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        executed
+    }
+}
+
+#[derive(Debug)]
+struct Callback {
+    keys: Vec<u32>,
+    callback: String,
```
Comment (PolyMeilex) [inline src/config/keyboard.rs:79]: Minor neat pick, could we keep FnPtr here, in order to keep the type info for as long as possible, an convert it to &str only when needed

**Review by PolyMeilex — CHANGES_REQUESTED** [review]
Besides the comment, it's great!

File: src/config/keyboard.rs (hunk 1 above)
Comment (dragonnn) [reply src/config/keyboard.rs:79]: I was testing with FnPtr it before, but it boils down that FnPtr isn't Send + Sync, so it is not trivial to save it state outside Rhai in lazy_static. At least I didn't find an option to do it.

**Comment by PolyMeilex** [comment]
Thanks

**Pull request merged**